ref(slack): Docs on Upgrading Slack (#1696)
getsentry/sentry-docs: src/collections/_documentation/workflow/integrations/global-integrations.md

    [{% asset slack-alert-message.png %}]({% asset slack-alert-message.png @path %})
    

> #### Deleting the legacy project-based Slack integration

Author: I updated this part because I thought it would be confusing for people if they didn't know what the Slack plugin was, and I didn't want folks to think the should delete the integration and reinstall. They should be going through the upgrade flow

Once you configure the global Slack integration and Alert Rules, you can disable the old project-based Slack integration. You’ll need to go to each project that has it enabled and disable it. We recommend disabling the legacy project-based integration after setting up the global integration. If you're looking to upgrade your global integration, please refer to the information below under **Upgrading Slack**.

### Upgrading Slack

#### How do I know if I need to upgrade? 

> If you installed the global Slack integration on a Sentry organization prior to May 8, 2020, you'll need to upgrade your integration. If you're unsure whether or not that includes your organization, please visit your integrations page via the Organization Settings > Integrations. You will see a call to action for the Slack integration. (screenshot?) (Please note we are still rolling this out, if you don't see this yet you may be in a later wave)

Reviewer: ```suggestion
You need to upgrade if you installed the global Slack integration on a Sentry organization before May 8, 2020. Unsure whether or not that includes your organization? Please visit your integration's page via **Organization Settings > Integrations**. You will see a call to action for the Slack integration. 

Please note, we're still rolling this out. If you don't see the call to action yet, you may be in a later group of adopters

(screenshot?) 
```

> 

Author: I'm not sure the best way to phase this but basically we want to have these docs available to link in the upgrade flow incase people have more questions. However, at first we will only be rolling it out to about ~5% of users that need to upgrade. We should have it available for all users (that have an outdated slack app) relatively soon but the idea was to minimize the impact incase people have a lots of questions/getting really confused etc. If we start with only 5% we can update wording/fix whats not clear for people going forward.

> If you installed the Slack integration on a Sentry organization after May 8, 2020, your integration will already be up to date and no upgrade is needed.

Reviewer: ```suggestion
If you installed the Slack integration on a Sentry organization after May 8, 2020, your integration will be up-to-date, and no upgrade is needed.
```

#### Why do I need to upgrade?

> We built our previous Slack integration on top of Slack's workspace apps. Unfortunately, not long after, Slack deprecated the workspace apps - more details on what went into that decision on [their blog](https://medium.com/slack-developer-blog/an-update-on-workspace-apps-aabc9e42a98b). While we've been able to maintain the current integration in the past, there are some upcoming deprecations to Slack APIs our workspace app relies on that require the move to their newly supported [bot app](maybe link to this) framework.

Reviewer: ```suggestion
Sentry built its previous Slack integration on top of Slack's workspace apps. Unfortunately, Slack deprecated the workspace apps shortly afterward. You can find more details about that decision on [the Slack blog](https://medium.com/slack-developer-blog/an-update-on-workspace-apps-aabc9e42a98b). 

While we were able to maintain the integration in the past, our workspace app relies on Slack APIs that will soon deprecate. We now need to move to Slack's newly supported [bot app](maybe link to this) framework.
```

#### What changes after my integration is upgraded?

> Previously, workspace apps allowed you to pick which channels you gave us access to, whether that was all public channels, or specific channels which could be either public or private.  Bot apps in Slack work differently. By default, the Sentry bot will be able to post alerts in public channels, but in order to be able to post to private channels, you'll need to add the bot to that specific channel.

Reviewer: ```suggestion
Previously, Slack workspace apps allowed you to give Sentry access to whichever channels you chose, including public or specific channels -- which could be either public or private. 

Bot apps in Slack work differently. By default, the Sentry bot will be able to post alerts in public channels, but to post in private channels, you'll need to add the bot to that specific channel.
```

> In terms of your alert rule configurations, if your current Slack integration only uses public channels then nothing needs to change after you upgrade.

Reviewer: ```suggestion
In terms of your alert rule configurations, if your current Slack integration only uses public channels, then nothing needs to change after you upgrade.
```

> Going forward, if you want to add private channels to alert rules, you'll need to add the Sentry bot to the channel first before making the rule in Sentry. 

Reviewer: ```suggestion
Going forward, if you want to add private channels to alert rules, you'll need to add the Sentry bot to the channel first before making the Sentry rule.
```

> If you do have any private channels currently used in your alert rules, you'll need to add the Sentry bot once you've upgraded in order for the alerts to continue to work. You'll have a chance in the upgrade flow to review which private channels, if any, are being used. We'll also send a message to each of those channels once you've upgrade to remind you to add the Sentry bot to that channel.

Reviewer: ```suggestion
If you have any private channels currently used in your alert rules, for the alerts to work, you'll need to add the Sentry bot after you've upgraded. In the upgrade flow, you'll have a chance to review which private channels you're using. Sentry will send a message to each channel, once you've upgraded, as a reminder to add the Sentry bot to that channel.
```

#### How do I upgrade?

1. Go to your Slack configurations page in Sentry.

>     [screenshot]

Author: All screenshots here are waiting on copy from Molly, and then for me to update in sentry to use the copy Molly has written. I can go over the flow for whoever is interested if you need more information.

2. Click **Upgrade Now**.
    [screenshot]

3. Click **Continue**.
    [screenshot]

4. If you have any private channels in use, you'll see them listed here. Click **Upgrade**.
    [screenshot]

#### When do I need to migrate by?

> All Sentry organizations will have until October 1, 2020 to upgrade their Slack integration. If you have any questions or concerns, please email us at partners@sentry.io.

Author: We are going to use `partners@sentry.io` for all 'support' links in the initial stages of the migration/upgrade. This may change later but for now we want to keep a close eye on folks and also to be able to respond quickly if there are any questions/concerns

> All Sentry organizations will have until October 1, 2020 to upgrade their Slack integration. If you have any questions or concerns, please email us at partners@sentry.io.

Reviewer: ```suggestion
All Sentry organizations will have until October 1, 2020, to upgrade their Slack integration. If you have any questions or concerns, please email us at [partners@sentry.io](mailto:partners@sentry.io).
```

    [{% asset slack-alert-message.png %}]({% asset slack-alert-message.png @path %})
    
#### Deleting the legacy project-based Slack integration

We recommend disabling the legacy project-based integration after setting up the global integration.

Once you configure the global Slack integration and Alert Rules, you can disable the old project-based Slack integration. Go to each project that has the old project-based Slack enabled and disable it. If you're looking to upgrade your global integration, please refer to the information below under **Upgrading Slack**.

### Upgrading Slack

#### How do I know if I need to upgrade? 

You need to upgrade if you installed the global Slack integration on a Sentry organization before May 8, 2020. Unsure whether or not that includes your organization? Please visit your integration's page via **Organization Settings > Integrations**. You will see a call to action for the Slack integration. 

[{% asset slack-migration-integration-directory.png %}]({% asset slack-migration-integration-directory.png @path %})

> Please note we are still rolling this out, if you don't see this yet you may be in a later wave.

Reviewer: ```suggestion
Please note Sentry is still rolling this out. If you don't see the call to action yet, you may be in a later wave of adopters.
```

If you installed the Slack integration on a Sentry organization after May 8, 2020, your integration will be up-to-date, and no upgrade is needed.

#### Why do I need to upgrade?

Sentry built its previous Slack integration on top of Slack's workspace apps. Unfortunately, Slack deprecated the workspace apps shortly afterward. You can find more details about that decision on [the Slack blog](https://medium.com/slack-developer-blog/an-update-on-workspace-apps-aabc9e42a98b). 

> While we were able to maintain the integration in the past, our workspace app relies on Slack APIs that will soon deprecate. We now need to move to Slack's newly supported bot app framework.

Reviewer: ```suggestion
While Sentry was able to maintain the integration in the past, our workspace app relies on Slack APIs that will soon deprecate. We now need to move to Slack's newly supported bot app framework.
```

#### What changes after my integration is upgraded?

Previously, Slack workspace apps allowed you to give Sentry access to whichever channels you chose, including public or specific channels -- which could be either public or private.

> Bot apps in Slack work differently. By default, the Sentry bot will be able to post alerts in public channels, but in order to be able to post to private channels, you'll need to add the bot to that specific channel.

Reviewer: ```suggestion
Bot apps in Slack work differently. By default, the Sentry bot will be able to post alerts in public channels, but to be able to post to private channels, you'll need to add the bot to that specific private channel.
```

> In terms of your alert rule configurations, if your current Slack integration only uses public channels, then nothing needs to change after you upgrade.

Reviewer: ```suggestion
In terms of your alert rule configurations, if your current Slack integration only uses public channels, nothing needs to change after you upgrade.
```
